topdown/encoding.go

```diff
@@ -235,9 +247,21 @@ func builtinYAMLUnmarshal(a ast.Value) (ast.Value, error) {
 	return ast.InterfaceToValue(val)
 }
 
+func builtinYAMLIsValid(a ast.Value) (ast.Value, error) {
+	str, err := builtins.StringOperand(a, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	var x interface{}
+	err = ghodss.Unmarshal([]byte(str), &x)
```

Comment:
🤔   I wonder how accurate this unmarshal call is for determining if some yaml is valid? I know something like `!!binary abc: foo` will return an error while being valid yaml, not sure the extent to which it supports everything, but I know it is somewhat limited by what it can/can't unmarshal into golang containers.

I don't think we should change how `is_valid` gets implemented, since it is a nice test for using the unmarshal builtin, but maybe it deserves a note in the docs?

Reply:
That's a good point.  The sole use case I have for this builtin is using it in front of `yaml.unmarshal` so it'll handle bad input gracefully.  I'll amend the docs to not it supports the same subset of YAML that `yaml.unmarshal` supports?